google/resource-snippets/cloudkms-v1/alternatives/krm/Kptfile

```diff
@@ -0,0 +1,23 @@
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+metadata:
+  name: cloudkms
+packageMetadata:
+  shortDescription: configure cloudkms
+openAPI:
+  definitions:
+    io.k8s.cli.setters.deployment:
+      description: deployment name
+      x-k8s-cli:
+        setter:
+          name: deployment
+          value: ${DEPLOYMENT?}
+          isSet: true
+    io.k8s.cli.substitutions.deployment-sub:
+      x-k8s-cli:
+        substitution:
+          name: deployment-sub
+          pattern: ${deployment}cryptokey
```

Comment:
This `${deployment}cryptokey` should be `${deployment}-cryptokey` to reflect the [TF](https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/master/google/resource-snippets/cloudkms-v1/alternatives/tf/cloudkms.tf#L17) example and [DM](https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/master/google/resource-snippets/cloudkms-v1/kms.jinja#L25). Otherwise `kpt cfg set . deployment d1` will produce KMSCryptoKey.name as `d1cryptokey`, not `d1-cryptokey`.

You need to delete this substitution and use something like `kpt cfg create-subst . deployment-sub --field-value '${DEPLOYMENT?}-cryptokey' --pattern '${deployment}-cryptokey'`

Also, I just learned this but you can see your substitutions via `kpt cfg list-setters --include-subst . `